addons/binding/org.openhab.binding.avmfritz/src/main/java/org/openhab/binding/avmfritz/handler/BoxHandler.java

```diff
@@ -174,6 +185,21 @@ private void updateThingFromDevice(Thing thing, DeviceModel device) {
                 updateState(channelPower.getUID(), new DecimalType(device.getPowermeter().getPower()));
             }
             if (device.isSwitchableOutlet() && device.getSwitch() != null) {
+                Channel channelMode = thing.getChannel(CHANNEL_MODE);
+                if (channelMode != null) {
+                    updateState(channelMode.getUID(), new StringType(device.getSwitch().getMode()));
+                } else {
+                    logger.warn("Channel {} in thing {} does not exist, please recreate the thing", CHANNEL_MODE,
```

Comment:
@martinvw I updated my PR. I introduced a new method `updateThingChannelState` to handle the logic and clean up the code.